UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment.UnitTests/Helpers/FileSystemHelperTests.cs

```diff
@@ -53,5 +52,37 @@ public void Does_CreateFolder_Completed_When_Directory_Doesnot_Exists()
             A.CallTo(() => _fakefileSystem.Directory.CreateDirectory(filePath))
                             .MustHaveHappenedOnceExactly();
         }
+
+        [Test]
+        public void Does_GetFileMD5_Returns_FileDetails_With_Hash()
+        {
+            List<Common.Models.Fss.Request.FileDetail> fileDetails = new();
+
+            IEnumerable<string> fileNames = new List<string> { fileName };
+
+            IFileInfo fileInfo = _fakefileSystem.FileInfo.FromFileName(fileName);
+            A.CallTo(() => fileInfo.Name).Returns(fileName);
+
+            A.CallTo(() => _fakefileSystem.FileInfo.FromFileName(A<string>.Ignored)).Returns(fileInfo);
+
+            List<Common.Models.Fss.Request.FileDetail>? result = _fileSystemHelper.GetFileMD5(fileNames);
+
+            Assert.Multiple(() =>
+            {
+                Assert.That(result.Count, Is.EqualTo(1));
+                Assert.That(result.FirstOrDefault().FileName, Is.EqualTo(fileName));
+                Assert.That(result.FirstOrDefault().Hash, Is.Not.Null);
+            });
+
+        }
+
+        [Test]
+        public void Does_GetFiles_Executes_Successful()
```

Comment:
Test name suggests it's checking for successful execution, but assert on line 84 checks that the call only happens once. Either change the name to reflect what the test is testing, or change the assert to confirm it was a successful run.

Reply:
Test case name updated